piff/stats.py

```diff
@@ -0,0 +1,364 @@
+
+# Copyright (c) 2016 by Mike Jarvis and the other collaborators on GitHub at
+# https://github.com/rmjarvis/Piff  All rights reserved.
+#
+# Piff is free software: Redistribution and use in source and binary forms
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+
+"""
+.. module:: stats
+
+"""
+
+from __future__ import print_function
+import numpy as np
+
+def process_stats(config, logger):
+    """Parse the stats field of the config dict.
+
+    :param config:      The configuration dict.
+    :param logger:      A logger object for logging debug info.
+
+    :returns: an stats instance
+    """
+    import piff
+
+    if 'stats' not in config:
+        raise ValueError("config dict has no stats field")
+    config_stats_list = config['stats']
+
+    stats = []
+    for config_stats in config_stats_list:
+
+        if 'type' not in config_stats:
+            raise ValueError("config['stats'] has no type field")
+
+        # Get the class to use for the stats
+        # Not sure if this is what we'll always want, but it would be simple if we can make it work.
+        stats_class = eval('piff.' + config_stats.pop('type'))
```

Comment:
This is the way @rmjarvis has been doing this in all the other files, so I just cribbed from him...

I think the usecase was that you were done with the config after processing (ie you loaded it up in a script), but I agree that you could instead do config_stats['type'] and not modify config_stats. I think in some of the other cases (like input.py), it was done so that if there were not a 'type' in the dictionary, it returned a default value:

    input_handler_class = eval('piff.Input' + config_input.pop('type','Files'))